pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -29,9 +29,17 @@ import (
 	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
 )
 
-// endpoint maps to a single registered device plugin. It is responsible
+// Endpoint maps to a single registered device plugin. It is responsible
 // for managing gRPC communications with the device plugin and caching
 // device states reported by the device plugin.
+type Endpoint interface {
+	Run()
```

Comment:
Maybe add some comment to each function?

Reply:
+1

Reply:
Skipping for now, as we are making this interface private.